### PR TITLE
removed try_depth ref (mutable state)

### PR DESCRIPTION
**Pull request by Anukriti12**

### CODE REVIEW

File: ocamltest/main.ml [hunk 1]
```diff
@@ -121,13 +120,9 @@ let get_test_build_directory_prefix test_dirname =
   if test_dirname = "." then root
   else Filename.concat root test_dirname
 
-let tests_to_skip = ref []
-
-let init_tests_to_skip () =
-  tests_to_skip := String.words (Sys.safe_getenv "OCAMLTEST_SKIP_TESTS")
-
```
Comment (gasche) [inline ocamltest/main.ml]: This part appears to be unrelated to the PR, can you remove it from the patch?

File: lambda/simplif.ml [hunk 2]
```diff
@@ -294,38 +292,37 @@ let simplify_exits lam =
...
   | Lsend(k, m, o, ll, loc) ->
-      Lsend(k, simplif m, simplif o, List.map simplif ll, loc)
-  | Levent(l, ev) -> Levent(simplif l, ev)
-  | Lifused(v, l) -> Lifused (v,simplif l)
+      Lsend(k, simplif try_depth m, simplif try_depth o, List.map (simplif try_depth) ll, loc)
+  | Levent(l, ev) -> Levent(simplif try_depth l, ev)
+  | Lifused(v, l) -> Lifused (v,simplif try_depth l)
   in
-  simplif lam
+  simplif 0 lam
```
Comment (gasche) [inline lambda/simplif.ml]: I think this callsite would be more readable if `try_depth` was a labelled/keyword parameter (`~try_depth` instead of `try_depth`). (Same for `count`.)

File: lambda/simplif.ml [hunk 3]
```diff
@@ -294,38 +292,37 @@ let simplify_exits lam =
...
-      incr try_depth;
-      let l1 = simplif l1 in
-      decr try_depth;
-      Ltrywith(l1, v, simplif l2)
-  | Lifthenelse(l1, l2, l3) -> Lifthenelse(simplif l1, simplif l2, simplif l3)
-  | Lsequence(l1, l2) -> Lsequence(simplif l1, simplif l2)
-  | Lwhile(l1, l2) -> Lwhile(simplif l1, simplif l2)
+      let l1 = simplif (try_depth + 1) l1 in
+      Ltrywith(l1, v, simplif try_depth l2)
+  | Lifthenelse(l1, l2, l3) -> Lifthenelse(simplif try_depth l1, simplif try_depth l2, simplif try_depth l3)
```
Comment (gasche) [inline lambda/simplif.ml]: In general the compiler codebase has to stay under 80 columns wide (see the `check-typo` test that you can run locally: `./tools/check-typo-since trunk` from your branch). I expect that you will have to break new lines at many places in your patch.

**Review by gasche — COMMENTED** [review]
I think overall the change is a good idea, as two things were not so clear in the previous code:
- the business of running `incr try_depth; ...; decr try_depth`
- the question of whether `count` and `simplif` made independent use of the reference or if they would call each other at non-zero depths.

There remain details to sort out before this is in good shape.

**Comment by hhugo** [comment]
I believe similar change was done in https://github.com/ocaml/ocaml/pull/11615. I think this PR can be closed
